### PR TITLE
bpo-41048: mimetypes should read the rule file using UTF-8, not the locale encoding

### PR DESCRIPTION
**Pull request by srinivasreddy**
https://bugs.python.org/issue41048

### CODE REVIEW

**Review by corona10 — CHANGES_REQUESTED** [review]
@srinivasreddy 

Thanks for the contribution,
Please add the NEWS.d by using blurb :)

guide: https://devguide.python.org/committing/#what-s-new-and-news-entries

**Comment by srinivasreddy** [comment]
@corona10 Done

File: Misc/NEWS.d/next/Library/2020-06-20-10-16-57.bpo-41048.hEXB-B.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+"read_mime_types" function reads the rule file using UTF-8 encoding, not the locale encoding.
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2020-06-20-10-16-57.bpo-41048.hEXB-B.rst]:
```suggestion
:func:`mimetypes.read_mime_types` function reads the rule file using UTF-8 encoding, not the locale encoding.
```

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
Please add tests.

File: Lib/test/test_mimetypes.py [hunk 2]
```diff
@@ -67,6 +67,20 @@ def test_read_mime_types(self):
             mime_dict = mimetypes.read_mime_types(file)
             eq(mime_dict[".pyunit"], "x-application/x-unittest")
 
+        # bpo-41048: read_mime_types should read the rule file with 'utf-8' encoding.
+        # Not with locale encoding. _bootlocale has been imported because io.open(...)
+        # uses it.
+        with support.temp_dir() as directory:
+            data = "application/no-mans-land  Fran\u00E7ais"
+            file = pathlib.Path(directory, "sample.mimetype")
+            file.write_text(data)
```
Comment (serhiy-storchaka) [inline Lib/test/test_mimetypes.py]: `write_text()` uses the locale encoding by default. Either specify the encoding or use `write_bytes()`.

File: Lib/test/test_mimetypes.py [hunk 3]
```diff
@@ -67,6 +67,20 @@ def test_read_mime_types(self):
...
+            import _bootlocale
+            getpreferredencoding = _bootlocale.getpreferredencoding(False)
+            self.addCleanup(setattr, _bootlocale, 'getpreferredencoding', getpreferredencoding)
```
Comment (serhiy-storchaka) [inline Lib/test/test_mimetypes.py]: `_bootlocale.getpreferredencoding` should be a function.

I suggest to use `test.support.swap_attr()`. It is easier and safer.
```
with support.swap_attr(_bootlocale, 'getpreferredencoding', lambda do_setlocale=True: 'ASCII'):
    mime_dict = mimetypes.read_mime_types(file)
eq(mime_dict[".Français"], "application/no-mans-land")
```

**Comment by srinivasreddy** [comment]
@serhiy-storchaka done. Thanks a lot

**Review by serhiy-storchaka — APPROVED** [review]

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**